grape/general_graph.py

```diff
@@ -136,17 +136,23 @@ def load(self, filename):
         self.Father_mark = nx.get_node_attributes(self, 'Father_mark')
         self.condition = nx.get_edge_attributes(self, 'Father_cond')
 
-        self.From_to = nx.get_node_attributes(self, 'From_to')
+        self.Type = nx.get_node_attributes(self, 'Type')
 
-        self.services_FROM = []
-        for id, From_to in self.From_to.items():
-            if From_to == "SOURCE":
-                self.services_FROM.append(id)
+        self.services_SOURCE = []
+        self.services_HUB = []
+        self.services_USER = []
+        for id, Type in self.Type.items():
+            if Type == "SOURCE":
+                self.services_SOURCE.append(id)
+            elif Type == "HUB":
+                self.services_HUB.append(id)
+            elif Type == "USER":
+                self.services_USER.append(id)
 
-        self.services_TO = []
-        for id, From_to in self.From_to.items():
-            if From_to == "TARGET":
-                self.services_TO.append(id)
+        #self.services_TO = []
+        #for id, From_to in self.From_to.items():
+        #    if From_to == "TARGET":
+        #        self.services_TO.append(id)
```

Comment:
Please remove if these rows are useless

Reply:
I need to update the readme. I am doing it now